filters/imagefilter.go

```diff
@@ -48,25 +49,12 @@ func HandleImageResponse(ctx filters.FilterContext, f ImageFilter) {
 
 	rsp.Header.Del("Content-Length")
 
-	in := rsp.Body
-	r, w := io.Pipe()
-	rsp.Body = r
-
-	go handleImageTransform(w, in, f)
+	handleImageTransform(rsp, f)
 }
 
-func handleImageTransform(out *io.PipeWriter, in io.ReadCloser, f ImageFilter) error {
+func handleImageTransform(rsp *http.Response, f ImageFilter) error {
```

Comment:
i would do this so:
- accept a simple io.Reader and return a new io.Reader:

```
func handleImageTransform(r io.Reader, f ImageFilter) (io.Reader, error) {
}
```

- internally readall from the reader like now from the body
- return the result as a bytes.Buffer (wrapping with nopcloser is best done in the HandleImageResponse method)